### PR TITLE
fix(api): align schemas with OpenAPI specs and switch search to POST

### PR DESCRIPTION
**Pull request by EdwardIrby**
## Summary

Aligns API schemas with the latest OpenAPI specs and migrates the search endpoint from GET to POST.

### Search API changes
- **GET → POST migration**: `fetchSearchResults` and `buildSearchRequest` now use POST with JSON body, matching the pattern used by research and contents endpoints
- **New fields**: `language`, `include_domains`, `exclude_domains`, `crawl_timeout` added to `SearchQuerySchema`
- **`livecrawl_formats`**: Changed from single enum string to array of enums per OpenAPI spec
- **`SEARCH_API_URL`**: Updated to `ydc-index.io/v1/search` to match OpenAPI spec
- **Client-side validation**: Rejects combining `include_domains` and `exclude_domains` (server returns 422)

### Research API changes
- **`snippets`**: Made optional in `ResearchSourceSchema` (only `url` is required per spec)
- **`formatResearchResponse`**: Added optional chaining for `source.snippets`

### Test updates
- All search tests updated for POST body assertions instead of query params
- New MCP `register-search-tool.spec.ts` with empty/success/error path coverage
- New research test for undefined snippets edge case
- Schema validation tests updated with new field conformance checks

### References
- Search spec: https://you.com/specs/openapi_search_v1.yaml
- Research spec: https://you.com/specs/openapi_research.yaml

### CODE REVIEW

**Review by itsakhilyou — COMMENTED** [review]
does doing away with `https://api.you.com/v1/agents/search` mean no more free search?

File: packages/api/src/search/search.utils.ts [hunk 1]
```diff
@@ -15,29 +15,22 @@ export const fetchSearchResults = async ({
...
   }
 
-  url.search = searchParams.toString()
-
   const options = {
-    method: 'GET',
+    method: 'POST',
     headers: new Headers({
       ...customHeaders,
       'X-API-Key': YDC_API_KEY || '',
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/api/src/search/search.utils.ts:24]: empty string is desired

[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/api/src/search/search.utils.ts:24]: Intentional — empty string lets the server return the appropriate error (403) rather than a client-side error, which is the desired behavior for this endpoint. See EdwardIrby's reply below.

File: packages/mcp/src/search/tests/register-search-tool.spec.ts [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        page_age: '2025-01-01T00:00:00',
+        authors: [],
+      },
+    ],
+    news: [],
+  },
+  metadata: { search_uuid: 'test', query: 'test', latency: 0.1 },
+}
+
+let mockFetchResponse: SearchResponse | Error = emptyResponse
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/mcp/src/search/tests/register-search-tool.spec.ts:31]: Fixed in aff9d50 — added `beforeEach` to reset `mockFetchResponse` to `emptyResponse` before each test.

File: packages/mcp/src/search/tests/register-search-tool.spec.ts [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+  registerSearchTool({ mcp: server, YDC_API_KEY: 'test-key', getUserAgent: () => 'test-agent' })
+
+  const [clientTransport, serverTransport] = InMemoryTransport.createLinkedPair()
+  await server.connect(serverTransport)
+
+  const client = new Client({ name: 'test-client', version: '0.0.0' })
+  await client.connect(clientTransport)
+
+  return { client, server }
+}
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/mcp/src/search/tests/register-search-tool.spec.ts:57]: Fixed in aff9d50 — `setupMcpClient` now returns a `cleanup` function that closes both client and server. Called in `afterEach`.

File: packages/api/src/search/tests/search.schema-validation.spec.ts [hunk 4]
```diff
@@ -219,21 +213,41 @@ describe('SearchQuerySchema conforms to live OpenAPI spec', () => {
...
+    const schemaLanguages = SearchQuerySchema.shape.language.unwrap().options
+    expect(schemaLanguages.length).toBe(51)
+  })
+
+  test('include_domains accepts valid arrays', () => {
+    expect(() => SearchQuerySchema.parse({ query: 'test', include_domains: ['you.com', 'example.com'] })).not.toThrow()
+  })
+
+  test('exclude_domains accepts valid arrays', () => {
+    expect(() => SearchQuerySchema.parse({ query: 'test', exclude_domains: ['spam.com'] })).not.toThrow()
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/api/src/search/tests/search.schema-validation.spec.ts:248]: Fixed in aff9d50 — now fetches `spec.components.schemas.CrawlTimeout` and compares dynamically, matching the pattern used by `count`.

File: packages/api/src/search/tests/search.schema-validation.spec.ts [hunk 5]
```diff
@@ -219,21 +213,41 @@ describe('SearchQuerySchema conforms to live OpenAPI spec', () => {
...
+    expect(specCount.minimum).toBeDefined()
+    expect(specCount.maximum).toBeDefined()
 
     const schemaCount = SearchQuerySchema.shape.count.unwrap()
-    expect(schemaCount.minValue).toBe(countParam?.schema.minimum)
-    expect(schemaCount.maxValue).toBe(countParam?.schema.maximum)
+    expect(schemaCount.minValue).toBe(specCount.minimum)
+    expect(schemaCount.maxValue).toBe(specCount.maximum)
+  })
+
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/api/src/search/tests/search.schema-validation.spec.ts:233]: Fixed in aff9d50 — now fetches the live spec and compares the language enum values directly (same as country/safesearch). Removed the hardcoded `51`.

File: .factory/settings.json [hunk 6]
```diff
@@ -0,0 +1,5 @@
+{
+  "enabledPlugins": {
+    "core@factory-plugins": true
+  }
+}
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply .factory/settings.json]: Fixed in aff9d50 — removed `.factory/settings.json` from tracking and added `.factory/` to `.gitignore`.

File: packages/api/src/search/search.utils.ts [hunk 7]
```diff
@@ -15,29 +15,22 @@ export const fetchSearchResults = async ({
...
-  const url = new URL(SEARCH_API_URL)
-
-  const searchParams = new URLSearchParams()
-
-  // Append all query parameters
-  for (const [name, value] of Object.entries(searchQuery)) {
-    if (value !== undefined && value !== null) {
-      searchParams.append(name, `${value}`)
-    }
+  if (searchQuery.include_domains && searchQuery.exclude_domains) {
```
[the remark this answers is not in the document]
Comment (EdwardIrby) [reply packages/api/src/search/search.utils.ts]: Fixed in aff9d50 — extracted `validateSearchQuery` helper in `search.schemas.ts` and both `fetchSearchResults` and `buildSearchRequest` now call it. Also removed the dead `queryParams` field from `DryRunResult` and updated the TSDoc.

**Comment by EdwardIrby** [comment]
Re: @itsakhilyou's question about free search — good catch. Reverted `SEARCH_API_URL` back to `https://api.you.com/v1/agents/search` in aff9d50. The proxied route is preserved for free tier support.
